ls1intum/Artemis
# `Programming exercises`: Improve usability for grading configuration

### Checklist
#### General



#### Client

### Motivation and Context


Adds the option to collapse and expand all test cases so that grading configuration is possible like it was in the prior interface with only test cases.

### Description

- Adds two buttons to collapse and expand all test cases
- Adds translations for test/task types
- Make weight, bonus points and bonus multiplier columns smaller and replaces their headings with icons
- Adds numeration of tasks and tests
- Fixes an issue where inactive test cases are also included in the task weight

### Steps for Testing

Prerequisites:
- 1 Programming Exercise

1. Go to the grading configuration e.g. `ARTEMISURL/course-management/4/programming-exercises/14/grading/test-cases`
2. Click the buttons and see if tasks collapse/expand correctly
3. Toggle tasks individually by clicking the arrow heads and try out the added buttons.

### Review Progress






### Screenshots



![image](https://user-images.githubusercontent.com/25489463/230727893-df43e566-0cce-4e89-aa6f-adcdf7768c5e.png)
![image](https://user-images.githubusercontent.com/25489463/230727901-98ee82c6-1f2d-4ee2-a75b-508d8bc067a2.png)
